microshift.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+export LC_ALL=C.UTF-8
+export LANG=C.UTF-8
+
+source tools.sh
+source snc-library.sh
+
+BUNDLE_TYPE="microshift"
+INSTALL_DIR=crc-tmp-install-data
+CRC_VM_NAME=${CRC_VM_NAME:-crc-microshift}
+BASE_DOMAIN=${CRC_BASE_DOMAIN:-testing}
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/$ARCH/clients/ocp-dev-preview}
+OPENSHIFT_RELEASE_VERSION=4.12.0-ec.4
+
+if grep -q -i "release 9" /etc/redhat-release
+then
+  echo "Check if system is registered"
+  # Check the subscription status and register if necessary
+  if ! sudo subscription-manager status >& /dev/null ; then
+     sudo subscription-manager register
+  fi
+else
+  echo "This script only works for RHEL-9"
+fi
+
+run_preflight_checks ${BUNDLE_TYPE}
+rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR}
+
+# Download the oc binary for specific OS environment
+download_oc
+OC=./openshift-clients/linux/oc
+
+# Generate a new ssh keypair for this cluster
+# Create a 521bit ECDSA Key
+rm id_ecdsa_crc* || true
+ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+
+# Configure host (RHEL-9) with required internal repo
```

Comment:
> internal 

ouch. nothing public?

Reply:
@gbraad not at this moment.

Reply:
I'd just stick with rhel8 for now to be closer to what microshift people test daily (my understanding is that they focus on rhel8?)

Reply:
@cfergeau as of now with 4.12 branch is locked they are going to start with RHEL-9 it is just matter of time when those internal repo become public. Also with rhel8 I need to again put a patch to upgrade the kernel to rhel-9 to make it work with M1.

Reply:
Upgrading to the rhel-9 kernel for M1 has been working well enough so far, are there microshift-specific issues which prevent us from doing the same for these bundles?
I'd much prefer to first let microshift do the full switch to rhel9. Once they have switched, we also start using rhel9 if this did not happen automatically.

Reply:
I didn't tried to have rhel-9 kernel upgrade on RHEL-8 and with 4.12 there are already early dev-preview (internal) with RHEL-9 so openshift bits are already tested with it and works as expected. That's the reason I directly went with RHEL-9 instead RHEL-8.

Reply:
I'd expect Microshift people to know best what works well for them, I don't understand why we need to second guess them and decide on our own that rhel9 is a better pick for testing.